phpunit/functional/UserTest.php

```diff
@@ -1012,6 +1012,84 @@ public function testGetFriendlyName($input, $rawname)
         $this->assertSame($rawname, $user->getFriendlyName());
     }
 
+    public function testGetFriendlyNameFields()
+    {
+        global $DB;
+
+        // Create test users with different combinations of name/firstname
+        $users_data = [
+            // User with both firstname and realname
+            [
+                'name' => 'user1',
+                'realname' => 'Doe',
+                'firstname' => 'John',
+                'expected' => 'Doe John' // Or 'John Doe' depending on names_format config
+            ],
+            // User with only realname
+            [
+                'name' => 'user2',
+                'realname' => 'Smith',
+                'firstname' => '',
+                'expected' => 'Smith'
+            ],
+            // User with only firstname
+            [
+                'name' => 'user3',
+                'realname' => '',
+                'firstname' => 'Alice',
+                'expected' => 'Alice'
+            ],
+            // User with neither realname nor firstname
+            [
+                'name' => 'user4',
+                'realname' => '',
+                'firstname' => '',
+                'expected' => 'user4'
+            ]
+        ];
+
+        $this->login();
+
+        // Check names_format to adjust expected results if necessary
+        $config = \Config::getConfigurationValues('core');
+        if ($config['names_format'] == User::FIRSTNAME_BEFORE) {
```

Comment:
We should know what the configuration is; an dnot rely here on a condition.
I guess you can just add the format in expected data array so all cases are tested.